dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/WeightedEncoding.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.dialogue.serde;
+
+import com.palantir.logsafe.Preconditions;
+import java.util.Objects;
+
+/**
+ * Represents an encoding with a weight for <code>Accept</code> types.
+ * Note that the weight may not be applied to the Accept header, rather
```

Comment:
Why lock ourselves out from applying q-factor to accept headers?

Reply:
My goal was to make it clear that it doesn't necessarily use q-factor values, but could in the future.